src/huggingface_hub/hf_api.py

```diff
@@ -3792,32 +3792,6 @@ def create_commit(
             f" {nb_deletions} deletion(s)."
         )
 
-        # If updating a README.md file, make sure the metadata format is valid
-        # It's better to fail early than to fail after all the files have been uploaded.
-        for addition in additions:
-            if addition.path_in_repo == "README.md":
-                with addition.as_file() as file:
-                    response = get_session().post(
-                        f"{ENDPOINT}/api/validate-yaml",
-                        json={"content": file.read().decode(), "repoType": repo_type},
-                        headers=headers,
-                    )
```

Comment:
This part has to be kept in `create_commit` as well with something like that:

```py
for addition in additions:
    if addition.path_in_repo == "README.md":
        with addition.as_file() as file:
            content = file.read().decode()
        self._validate_yaml(content, ...)
```

Not all commits are created with `upload_folder` so we cannot assume the check has already been done.

Reply:
Added back in, thanks.